posts/2020-03-31-Rust-survey-2019.md

```diff
@@ -0,0 +1,216 @@
+
+![Why have you never used Rust?](images/2020-03-RustSurvey/5-Why-Never-Used-Rust.png)
+
+For users who indicated they had never used Rust before, most indicated either "I haven't learned Rust yet, but I want to" or "My company doesn't use Rust" - again pointing to adoption as the main hurdle.
+
+![I would use Rust more often if...](images/2020-03-RustSurvey/20-Would-use-rust-more-often-if.png)
+
+We also asked users what would lead them to use Rust more often. Most indicated they would use Rust more if their company adopted it, if Rust had more libraries that they need, and if IDE support was better. The most common reasons after those pointed to a need to improve the learning curve and interoperability. 
+
+As adoption seemed to be the biggest problem preventing some respondents from using Rust, let's dive deeper into it. 
+
+## Rust Adoption - a Closer Look
+
+First, we asked what would make Rust more appealing to your company and convince them to adopt it.
+
+![How can we make Rust more appealing to your company](images/2020-03-RustSurvey/21-Rust-more-appealing.png)
+
+Several users gave specific examples:
+* "Smoothest learning curve as possible, as a small business even 4-6 weeks to become productive is a lot to ask"
+* "Higher market penetration"
+* "More stable libraries"
+* "A full-stack web framework like Rails, Django and Phoenix"
+* "Better documentation, more examples, recommendation on what crates to use"
+* "More emphasis on how it is a safer alternative to C or C++ (and really should be the default usually).”
+* "Improve compile times. Compiling development builds at least as fast as Go would be table stakes for us to consider Rust. (Release builds can be slow."
+* "Better platform support"
+* "Security and performance, cost efficient and "green" (low carbon footprint) language"
+* "Embedded development targeting ARM"
+* "Better GUI framework, similar to Qt or directly using Qt via bindings."
+
+Most indicated that Rust maturity - such as more libraries and complete learning resources and more mature production capabilities - would make Rust more appealing.
+
+Let's take a closer look at each of these, starting with the need for more mature libraries.
+
+## Libraries - a Closer Look
+
+When we asked users what libraries they consider critical to the Rust ecosystem, these were the top ten responses:
+* serde
+* rand
+* tokio
+* async
+* clap
+* regex
+* log
+* futures
+* hyper
+* lazy_static
+
+We also asked how many dependencies users were using were 1.0 or above.
+* 0.8% indicated "All"
+* 6.7% indicated "Most"
+* 65.9% indicated "Some"
+* 5.2% indicated "None"
+* 21.4% indicated "I don't know"
+
+[Add COPY/analysis here]
```

Comment:
note: in particular, we should try to tie the libraries above -- I think all of which, with the exception of serde/clap/regex/lazy_static, are very low level -- you would generally want to wrap them in something else to get the same nicety of a framework-like experience in another language.